src/DidCache.ts

```diff
@@ -1,16 +1,338 @@
+import { DidDocument } from '@decentralized-identity/did-common-typescript';
+import * as Base58 from 'bs58';
+import { Cas } from './Cas';
+import Multihash from './Multihash';
+import { WriteOperation, OperationType } from './Operation';
 import Transaction from './Transaction';
-import { WriteOperation } from './Operation';
 
 /**
- * Interface for caching all the processed DIDs and related states.
- * This interface is mainly useful for creating a mock instances for testing purposes.
+ * VersionId identifies the version of a DID document. We use the hash of the
+ * operation that produces a particular version of a DID document as its versionId.
+ * This usage is guaranteed to produce unique VersionId's since the operation contains
+ * as one of its properties the previous VersionId. Since the operation hash is
+ * just a string we alias VersionId to string.
+ *
+ * With this usage, the operation hash serves two roles (1) an identifier for an operation
+ * (2) an identifier for the DID document produced by the operation. In the code below,
+ * we always use VersionId in places where we mean (2) and an OperationHash defined below
+ * when we mean (1).
  */
-export interface DidCache {
```

Comment:
Please keep this interface and perhaps rename the implementation class to InMemoryDidCache (until it's not), my upcoming unit test still depends on the mock DidCache.

Reply:
Sure, retained the old interface and added an implementation cache.